pkg/agent/plugin/nodeattestor/k8s/sat.go

```diff
@@ -0,0 +1,139 @@
+package k8s
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"sync"
+
+	"github.com/hashicorp/hcl"
+	uuid "github.com/satori/go.uuid"
+	"github.com/spiffe/spire/pkg/common/plugin/k8s"
+	"github.com/spiffe/spire/proto/agent/nodeattestor"
+	"github.com/spiffe/spire/proto/common"
+	spi "github.com/spiffe/spire/proto/common/plugin"
+	"github.com/zeebo/errs"
+)
+
+const (
+	pluginName = "k8s_sat"
+
+	defaultTokenPath = "/run/secrets/kubernetes.io/serviceaccount/token"
+)
+
+var (
+	satError = errs.Class("k8s-sat")
+)
+
+type SATAttestorConfig struct {
+	TokenPath string `hcl:"token_path"`
+}
+
+type satAttestorConfig struct {
+	trustDomain string
+	tokenPath   string
+}
+
+type SATAttestorPlugin struct {
+	mu     sync.RWMutex
+	config *satAttestorConfig
+
+	hooks struct {
+		newUUID func() string
+	}
+}
+
+var _ nodeattestor.Plugin = (*SATAttestorPlugin)(nil)
+
+func NewSATAttestorPlugin() *SATAttestorPlugin {
+	p := &SATAttestorPlugin{}
+	p.hooks.newUUID = func() string {
+		return uuid.NewV4().String()
+	}
+	return p
+}
+
+func (p *SATAttestorPlugin) FetchAttestationData(stream nodeattestor.FetchAttestationData_PluginStream) error {
+	config, err := p.getConfig()
+	if err != nil {
+		return err
+	}
+
+	uuid := p.hooks.newUUID()
+
+	token, err := loadTokenFromFile(config.tokenPath)
+	if err != nil {
+		return satError.New("unable to get token value: %v", err)
```

Comment:
nit: perhaps `"unable to read token from %v: %v", config.tokenPath, err` or something similar will provide more information?